css3 syntax encoding tests

r12a opened the pull request:
<!-- Reviewable:start -->

<!-- Reviewable:end -->

r12a committed:
css3 syntax encoding tests

w3c-bots commented:
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5549/)

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/221490218)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 674a82203fea4c2882cca582736859b34adc7699
Using browser at version BuildID 20170412100254; SourceStamp f40e24f40b4c4556944c762d4764eace261297f5
Starting 10 test iterations
No tests run.

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/221490224)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 674a82203fea4c2882cca582736859b34adc7699
Using browser at version 59.0.3067.0 dev
Starting 10 test iterations
No tests run.

r12a commented:
Can someone help me move these tests forward? I don't believe that the tabs or trailing spaces cause any problems (we have run the tests at https://www.w3.org/International/tests/repo/results/css-input-byte-stream and all pass (apart from 2 just for IE). I don't really want to spend the time to remove them because many of these files are in non-utf-8 encodings, which means one has to be careful.

The duplicate file names are, i assume, those for the tests mentioned in https://github.com/w3c/web-platform-tests/issues/5490, which are tests for CSS 2.1 pre-errata, and should be removed.

gregwhitworth commented:
I think @tabatkins and @SimonSapin are the best to review this. That said, if they haven't reviewed it by next Wednesday, I'll take a look. Thanks for the reminder @r12a

r12a commented:
ping?

gregwhitworth commented:
Hey, I asked @tabatkins to review these at the last F2F. I'll ping him again about it.

tabatkins commented on css/css-syntax-3/tokenizing-and-parsing/input-byte-stream/i18n-css-syntax-bytes-017.html:
```diff
@@ -0,0 +1,49 @@
+
+The class ids will therefore only match if the CSS is read as UTF-8.
+
+The style sheet has a UTF-8 signature at the start, but the HTTP header for the style sheet says ISO 8859-1. If the HTTP header overrides the @charset rule, the CSS will not be read as UTF-8, the selector will not match the class name, and the test will fail.
```
This line implies the stylesheet has a @charset; it actually has a BOM.

tabatkins commented on css/css-syntax-3/tokenizing-and-parsing/input-byte-stream/i18n-css-syntax-bytes-018.html:
```diff
@@ -0,0 +1,51 @@
+
+p { color: gray; margin-top: 3em; font-size: 80%; }
+</style>
+<link rel="stylesheet" type="text/css" href="support/bom-charset15.css">
```
This stylesheet has a `Content-Type` of `text/html`. Per <https://drafts.csswg.org/css-cascade/#content-type>, this should cause it to be ignored, since this is a standards-mode document.  Thus this test isn't testing quite the right thing.

tabatkins commented on css/css-syntax-3/tokenizing-and-parsing/input-byte-stream/support/bom-charset15.css.headers:
```diff
@@ -0,0 +1 @@
+Content-Type: text/html
```
Why do most of these CSS files have a `Content-Type: text/html`? Per <https://drafts.csswg.org/css-cascade/#content-type>, this should cause them to be ignored (unless they're linked from a same-origin quirks-mode document, in which case we ignore the declared content-type).  This looks like a persistent bug across a bunch of tests.

tabatkins replied:
Why do most of these CSS files have a Content-Type: text/html? Per https://drafts.csswg.org/css-cascade/#content-type, this should cause them to be ignored (unless they're linked from a same-origin quirks-mode document, in which case we ignore the declared content-type). This looks like a persistent bug across a bunch of tests.

tabatkins reviewed (changes requested):
Looks good other than the Content-Type bug for most of the stylesheets.

r12a committed:
corrected header files

r12a committed:
i18n-css-syntax-bytes-019, corrected encoding of html file

r12a committed:
i18n-css-syntax-bytes-017, changed comment to say bom instead of @charset rule

r12a commented:
@tabatkins thanks for the review. I just added 3 commits addressing the issues you found, and a couple more i found while fixing 
1. some headers files were misnamed, 
2. the encoding of one html file was incorrect).  
With luck, we may be done now(?)

w3c-bots commented:
# Build ERRORED

Started: 2017-09-04 16:00:21
Finished: 2017-09-04 16:07:11


<h2>Failing Jobs</h2>
<ul>

<li>lint</li>

<li>build_css</li>

<li>chrome:unstable</li>

<li>safari:10.0</li>

<li>MicrosoftEdge:14.14393</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18163)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/271746243)

Ms2ger commented on css/css-syntax-3/tokenizing-and-parsing/input-byte-stream/i18n-css-syntax-bytes-001.html:
```diff
@@ -0,0 +1,47 @@
+<link rel='help' href='http://www.w3.org/TR/css3-syntax/#input-byte-stream'>
+<script src="../../../resources/testharness.js"></script>
+<script src="../../../resources/testharnessreport.js"></script>
+<link rel="stylesheet" type="text/css" href="../../resources/../testharness.css">
```
You should remove the references to `testharness.css` (especially because it's wrong).

Ms2ger commented on css/css-syntax-3/tokenizing-and-parsing/input-byte-stream/i18n-css-syntax-bytes-001.html:
```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
+<html  lang="en" >
+<head>
+<meta charset="ISO-8859-15"/>
```
Do we need both a meta element and a header?

r12a replied:
yes

Ms2ger commented on css/css-syntax-3/tokenizing-and-parsing/input-byte-stream/i18n-css-syntax-bytes-001.html:
```diff
@@ -0,0 +1,47 @@
+-->
+<script>
+test(function() {
+assert_equals(document.getElementById('box').offsetWidth, 100);
```
Would these tests be better as reftests?

r12a replied:
no

Ms2ger reviewed (changes requested):
Passing the lint is not optional. If you're insistent on keeping the whitespace issues, use `lint.whitelist`.

r12a committed:
css link removed
extra space at the end of one *comment* line removed from files to pass the ridiculous linting requirements

r12a commented:
Giving up on this.  If you want the tests, please contact me.
